secrets.go

```diff
@@ -0,0 +1,313 @@
+package lenses
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	keyvault "github.com/Azure/azure-sdk-for-go/services/keyvault/2016-10-01/keyvault"
+	azureAuth "github.com/Azure/go-autorest/autorest/azure/auth"
+	vaultapi "github.com/hashicorp/vault/api"
+	golog "github.com/kataras/golog"
+)
+
+var providerType SecretProvider
+
+// SecretProvider secret provider type
+type SecretProvider string
+
+const (
+	// Vault Hashicorp Vault
+	Vault SecretProvider = "VAULT"
+	// AzureKV Azure KeyVault
+	AzureKV SecretProvider = "AZURE_KV"
+	// Kubernetes Kubernetes secrets
+	Kubernetes SecretProvider = "KUBERNETES"
+)
+
+type vaultAppRoleCredentials struct {
+	RoleID   string
+	SecretID string
+}
+
+func loadSecrets(file string) ([]string, error) {
+	var secretVars []string
+
+	if file != "" {
+		golog.Infof("Loading variables from file [%s]", file)
+		lines, err := ReadLines(file)
+
+		for _, l := range lines {
+			if strings.HasPrefix(l, "SECRET_") {
+				secretVars = append(secretVars, strings.Replace(l, "SECRET_", "", -1))
+			}
+		}
+
+		if err != nil {
+			return secretVars, err
+		}
+
+	} else {
+		secretVars = getSecretVars()
+	}
+
+	if len(secretVars) == 0 {
+		golog.Warnf("No environment variables prefixed with [SECRET_] found or loaded from file [%s]", file)
+	}
+
+	return secretVars, nil
+}
+
+func getSecretVars() []string {
+	var secretVars []string
+	// get secret vars
+	golog.Info("Looking for secret environment variables")
+	vars := os.Environ()
+
+	for _, v := range vars {
+		if strings.HasPrefix(v, "SECRET_") {
+			golog.Infof("Found environment var [%s]", v)
+			secretVars = append(secretVars, strings.Replace(v, "SECRET_", "", -1))
+		}
+	}
+
+	return secretVars
+}
+
+func getVaultClient(server, token string) (*vaultapi.Client, error) {
+	client, err := vaultapi.NewClient(&vaultapi.Config{
```

Comment:
why not set the token here and you are setting it after the error check?

Reply:
I don't understand?

Reply:
here you set the token https://github.com/Landoop/lenses-go/pull/6/files#diff-f312e33df50115b7e158e79a3e55d96bR89


but if you have `VAULT_TOKEN` in environment file this is set automatically. https://github.com/hashicorp/vault/blob/master/api/client.go#L342